eng/common/scripts/Service-Level-Readme-Automation.ps1

```diff
@@ -0,0 +1,223 @@
+  $mgmtIndexReadme  = "$readmeBaseName-mgmt-index.md"
+  $clientPackageInfo = $packageInfos.Where({ 'client' -eq $_.Type }) | Sort-Object -Property Package
+  if ($clientPackageInfo) {
+    generate-markdown-table -readmeFolder $readmeFolder -readmeName "$clientIndexReadme" -packageInfo $clientPackageInfo -moniker $moniker
+  }
+  # TODO: we currently do not have the right decision on how we display mgmt packages. Will track the mgmt work in issue. 
+  # https://github.com/Azure/azure-sdk-tools/issues/3422
+  # $mgmtPackageInfo = $packageInfos.Where({ 'mgmt' -eq $_.Type }) | Sort-Object -Property Package
+  # if ($mgmtPackageInfo) {
+  #   generate-markdown-table -readmeFolder $readmeFolder -readmeName "$mgmtIndexReadme" -packageInfo $mgmtPackageInfo -moniker $moniker
+  # }
+  if (!(Test-Path (Join-Path $readmeFolder -ChildPath $serviceReadme))) {
+    create-metadata-table -readmeFolder $readmeFolder -readmeName $serviceReadme -moniker $moniker -msService $msService `
+      -clientTableLink $clientIndexReadme -mgmtTableLink $mgmtIndexReadme `
+      -serviceName $serviceName
+  }
+  else {
+    update-metadata-table -readmeFolder $readmeFolder -readmeName $serviceReadme -serviceName $serviceName -msService $msService
+  }
+}
+
+$fullMetadata = Get-CSVMetadata
+$monikers = @("latest", "preview")
+foreach($moniker in $monikers) {
+  $onboardedPackages = &$GetOnboardedDocsMsPackagesForMonikerFn `
+    -DocRepoLocation $DocRepoLocation -moniker $moniker
+  $csvMetadata = @()
+  foreach($metadataEntry in $fullMetadata) {
+    if ($metadataEntry.Package -and $metadataEntry.Hide -ne 'true') {
+      $pkgKey = GetPackageKey $metadataEntry
+      if($onboardedPackages.ContainsKey($pkgKey)) {
+        $jsonFileName = $pkgKey.Replace('@azure/', 'azure-')
+        $jsonFilePath = "$DocRepoLocation/metadata/$moniker/$jsonFileName.json"
+        if(!(Test-Path $jsonFilePath)){
+          $csvMetadata += $metadataEntry
+          continue
+        }
+        if (!($metadataEntry.PSObject.Members.Name -contains "DirectoryPath")) {
+          $metadataJsonFile = Get-Content $jsonFilePath -Raw | ConvertFrom-Json
+          Add-Member -InputObject $metadataEntry `
+          -MemberType NoteProperty `
+          -Name DirectoryPath `
+          -Value $metadataJsonFile.DirectoryPath
+        }
+        $csvMetadata += $metadataEntry
+      }
+    }
+  }
+  $packagesForService = @{}
+  $allPackages = GetPackageLookup $csvMetadata
+  foreach ($metadataKey in $allPackages.Keys) {
+    $metadataEntry = $allPackages[$metadataKey]
+    if (!$metadataEntry.ServiceName) {
+      LogWarning "Empty ServiceName for package `"$metadataKey`". Skipping."
+      continue
+    }
+    $packagesForService[$metadataKey] = $metadataEntry
+  }
+  $services = @{}
+  foreach ($package in $packagesForService.Values) {
+    if ($package.ServiceName -eq 'Other') {
+      # Skip packages under the service category "Other". Those will be handled
```

Comment:
Do these entries get handled later? I think we just ignore everything under `Other` in this case because there's not an interesting service-level overview for `Other`

Reply:
Yes, we ignored Other for now.